core/src/main/java/org/apache/iceberg/avro/PruneColumns.java

```diff
@@ -297,4 +298,19 @@ private static Schema copyRecord(Schema record, List<Schema.Field> newFields) {
   private static boolean isOptionSchemaWithNonNullFirstOption(Schema schema) {
     return AvroSchemaUtil.isOptionSchema(schema) && schema.getTypes().get(0).getType() != Schema.Type.NULL;
   }
+
+  // for primitive types, the visitResult will be null, we want to reuse the primitive types from the original
+  // schema, while for nested types, we want to use the visitResult because they have content from the previous
+  // recursive calls.
+  private static Schema copyUnion(Schema record, List<Schema> visitResults) {
```

Comment:
What will ```null``` in ```[null, int, string]``` be in record and visitResults here?

Reply:
It weill be both `null`.